store/sqlstore/user_store.go

```diff
@@ -381,6 +391,28 @@ func (us SqlUserStore) GetAllProfiles(offset int, limit int) store.StoreChannel
 	})
 }
 
+func substituteWhereClause(searchQuery string, options *model.UserGetOptions) string {
+	whereClause := ""
+	whereClauses := []string{}
+	if options.Role != "" {
+		whereClauses = append(whereClauses, fmt.Sprintf(" Users.Roles like'%%%v%%' ", options.Role))
```

Comment:
This looks like a possible SQL injection attack, as I can inject an arbitrary string through the API that ends up in `options.Role` which is then substituted into the database query string directly. I think the way to do it here is to use a named parameter for the value of the `LIKE` clause, and perhaps pass in to this function the map that gets given ultimately to the query in order to add that value to it.

Reply:
@grundleborg removed the concatenation and used a named Parameter. I am not sure if this is way you mentioned.  Kindly Let me know if I got your point right.